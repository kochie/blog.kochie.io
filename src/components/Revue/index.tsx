import React from 'react'
import Card from '../Card'

const Revue = () => {
  return (
    <div
      id="revue-embed"
      className="relative max-w-5xl mx-auto px-4 mb-0 pb-10 mt-10"
    >
      <Card>
        <form
          className="p-4 md:p-8 lg:p-14"
          action="http://newsletter.kochie.io/add_subscriber"
          method="post"
          id="revue-form"
          name="revue-form"
          target="_blank"
        >
          <h1 className="text-2xl mb-5">
            Like what you see...? <br />
            Find out when I sporadically scream into the void
          </h1>
          <div className="grid grid-cols-6 gap-6">
            <div className="col-span-6">
              <label
                htmlFor="member_email"
                className="block text-sm font-medium text-gray-700 dark:text-gray-100"
              >
                Email Address
              </label>
              <input
                className="text-black px-3 py-2 mt-1 block w-full rounded-md border-gray-300 shadow-sm sm:text-sm"
                placeholder="tony@stark.industries"
                type="email"
                name="member[email]"
                id="member_email"
              />
            </div>

            <div className="col-span-6 sm:col-span-3">
              <label
                htmlFor="member_first_name"
                className="block text-sm font-medium text-gray-700 dark:text-gray-100"
              >
                First Name <span className="optional">(Optional)</span>
              </label>
              <input
                className="text-black px-3 py-2 mt-1 block w-full rounded-md border-gray-300 shadow-sm sm:text-sm"
                placeholder="Tony"
                type="text"
                name="member[first_name]"
                id="member_first_name"
              />
            </div>
            <div className="col-span-6 sm:col-span-3">
              <label
                htmlFor="member_last_name"
                className="block text-sm font-medium text-gray-700 dark:text-gray-100"
              >
                Last Name <span className="optional">(Optional)</span>
              </label>
              <input
                className="text-black px-3 py-2 mt-1 block w-full rounded-md border-gray-300 shadow-sm sm:text-sm"
                placeholder="Stark"
                type="text"
                name="member[last_name]"
                id="member_last_name"
              />
            </div>
            <div className="col-span-6 md:col-span-1">
              <input
                type="submit"
                value="Subscribe"
                name="member[subscribe]"
                id="member_submit"
                className="w-full p-4 bg-orange-500 hover:bg-orange-600 text-white font-bold rounded transform duration-200 cursor-pointer"
              />
            </div>
            <div className="col-span-6 md:col-span-5 my-auto">
              By subscribing, you agree with Revue&apos;s{' '}
              <a
                target="_blank"
                rel="noreferrer"
                href="https://www.getrevue.co/terms"
                className="underline text-orange-500 hover:text-orange-600 transform duration-200"
              >
                Terms of Service
              </a>{' '}
              and{' '}
              <a
                target="_blank"
                rel="noreferrer"
                href="https://www.getrevue.co/privacy"
                className="underline text-orange-500 hover:text-orange-600 transform duration-200"
              >
                Privacy Policy
              </a>
              .
            </div>
          </div>
        </form>
      </Card>
    </div>
  )
}

export default Revue
