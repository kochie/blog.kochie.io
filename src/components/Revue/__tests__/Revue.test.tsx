import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'

import Revue from '..'

describe('Revue Component', () => {
  test('renders correctly', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(<Revue />)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
