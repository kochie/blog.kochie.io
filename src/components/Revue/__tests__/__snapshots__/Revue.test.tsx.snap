// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Component renders correctly 1`] = `
<div
  className="relative max-w-5xl mx-auto px-4 mb-0 pb-10 mt-10"
  id="revue-embed"
>
  <div
    className="shadow-2xl rounded-2xl transition ease-in-out duration-200 bg-gray-300 dark:bg-neutral-600 text-black dark:text-white w-full h-full z-20"
  >
    <form
      action="http://newsletter.kochie.io/add_subscriber"
      className="p-4 md:p-8 lg:p-14"
      id="revue-form"
      method="post"
      name="revue-form"
      target="_blank"
    >
      <h1
        className="text-2xl mb-5"
      >
        Like what you see...? 
        <br />
        Find out when I sporadically scream into the void
      </h1>
      <div
        className="grid grid-cols-6 gap-6"
      >
        <div
          className="col-span-6"
        >
          <label
            className="block text-sm font-medium text-gray-700 dark:text-gray-100"
            htmlFor="member_email"
          >
            Email Address
          </label>
          <input
            className="text-black px-3 py-2 mt-1 block w-full rounded-md border-gray-300 shadow-sm sm:text-sm"
            id="member_email"
            name="member[email]"
            placeholder="tony@stark.industries"
            type="email"
          />
        </div>
        <div
          className="col-span-6 sm:col-span-3"
        >
          <label
            className="block text-sm font-medium text-gray-700 dark:text-gray-100"
            htmlFor="member_first_name"
          >
            First Name 
            <span
              className="optional"
            >
              (Optional)
            </span>
          </label>
          <input
            className="text-black px-3 py-2 mt-1 block w-full rounded-md border-gray-300 shadow-sm sm:text-sm"
            id="member_first_name"
            name="member[first_name]"
            placeholder="Tony"
            type="text"
          />
        </div>
        <div
          className="col-span-6 sm:col-span-3"
        >
          <label
            className="block text-sm font-medium text-gray-700 dark:text-gray-100"
            htmlFor="member_last_name"
          >
            Last Name 
            <span
              className="optional"
            >
              (Optional)
            </span>
          </label>
          <input
            className="text-black px-3 py-2 mt-1 block w-full rounded-md border-gray-300 shadow-sm sm:text-sm"
            id="member_last_name"
            name="member[last_name]"
            placeholder="Stark"
            type="text"
          />
        </div>
        <div
          className="col-span-6 md:col-span-1"
        >
          <input
            className="w-full p-4 bg-orange-500 hover:bg-orange-600 text-white font-bold rounded transform duration-200 cursor-pointer"
            id="member_submit"
            name="member[subscribe]"
            type="submit"
            value="Subscribe"
          />
        </div>
        <div
          className="col-span-6 md:col-span-5 my-auto"
        >
          By subscribing, you agree with Revue's
           
          <a
            className="underline text-orange-500 hover:text-orange-600 transform duration-200"
            href="https://www.getrevue.co/terms"
            rel="noreferrer"
            target="_blank"
          >
            Terms of Service
          </a>
           
          and
           
          <a
            className="underline text-orange-500 hover:text-orange-600 transform duration-200"
            href="https://www.getrevue.co/privacy"
            rel="noreferrer"
            target="_blank"
          >
            Privacy Policy
          </a>
          .
        </div>
      </div>
    </form>
  </div>
</div>
`;
